### DISCUSSION TITLE
gf to goto file not working after the update to Snack.terminal

### DISCUSSION THREAD

**Discussion by m4kamran** [LazyVim/LazyVim#4741] [General]
Previously in the terminal I was able to jump to a file from the terminal using `gf` but it does not working anymore with Snack terminal and it also closes the running npm process.

Goto file works fine in the editor.

**Comment by dpetka2001** [comment]
Could you see if [this PR](https://github.com/folke/snacks.nvim/pull/45) solves your issue?

**Reply by m4kamran** [reply]
Yes. It fixes the go to file for me. Awesome. 

But I have another problem. I'm working in a multi project repo. And after toggling the terminal it opens in a current project rather than the root directory? Is it possible to have it open in the root directory by default?

**Reply by dpetka2001** [reply]
By default Snacks terminal opens in the root directory depending on how LSP resolves the `root_dir`. Snacks just uses the Neovim terminal but opened either in float or split depending on how you configure it. You can also pass directly in the command the current working directory that you want the terminal to open into, so something like `:lua Snacks.terminal(nil, { cwd = "your_own_logic_here" })` should achieve what you want.
